src/dlgConnectionProfiles.h

```diff
@@ -39,7 +39,7 @@ class dlgConnectionProfiles : public QDialog, public Ui::connection_profiles
     void accept() override;
 
 signals:
-    void signal_establish_connection(QString profile_name, int historyVersion);
+    void signal_load_profile(QString profile_name, int historyVersion, bool alsoConnect);
```

Comment:
I noticed this, but assumed it was added to the API for future use, and didn't want to _remove_ the possibility.

Reply:
:rofl: It is more likely some ancient artefact that has long been obsolete...

Reply:
I looked into it. It's ancient.
It's referenced as a complete feature in a Jan 2009 commit, but the oldest source code available (March 2009) doesn't use it. :laughing:

Reply:
We can leave it I suppose to a separate clean-up PR just to take it out...

Reply:
:beetle: Huston - we may have a problem.

This SIGNAL has been extended, from:
```
(void) signal_load_profile(QString, int)
```
to:
```
(void) signal_establish_connection(QString, int, bool)
```

but the SLOT it is connected to has been changed from:
```
(void) slot_connection_dlg_finished(QString, int)
```
to
```
(void) slot_connection_dlg_finished(QString, bool)
```

I think this will compile because the `int` from the SIGNAL can be converted to a `bool` but the wanted third argument will be silently dropped and will not be seen by the SLOT method - and thus it will always see a `0` from the call to `signal_load_profile(...) ` at the bottom of `mudlet::loadProfile(...)`, treat it as a `false` and never connect even if the user clicks the "Connect" button and not the "Load" one...

Reply:
Yep, you're totally right. @LiamLeFey is this something you can fix?